pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -545,6 +545,90 @@ def prompted_output_instructions(self) -> str | None:
     __repr__ = _utils.dataclasses_no_defaults_repr
 
 
+@dataclass
+class _ResolvedToolChoice:
+    """Provider-agnostic resolved tool choice.
+
+    This is the result of validating and resolving the user's `tool_choice` setting.
+    Providers should map this to their API-specific format.
+    """
+
+    mode: Literal['none', 'auto', 'required', 'specific']
+    """The resolved tool choice mode."""
+
+    tool_names: list[str] = field(default_factory=list)
```

Comment:
Instead of this class, we could pass around the `Literal['none', 'required', 'auto'] | list[str]` value that the user provides, and have `filter_tools` be a helper method right? I'm not sure a separate type is warranted here